compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -954,7 +954,11 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
 
         let new_obligation =
             self.mk_trait_obligation_with_new_self_ty(obligation.param_env, trait_pred_and_self);
-        if self.predicate_must_hold_modulo_regions(&new_obligation) {
+        if self.predicate_must_hold_modulo_regions(&new_obligation)
+        // FIXME expr and stmt have the same span if expr comes from expansion
+        // cc: https://github.com/rust-lang/rust/pull/147416#discussion_r2499407523
+        && stmt.span.eq_ctxt(tail_expr.span)
```

Comment:
Lol, of course. The spans are identical, so that's always true ^^' (I'm so hung up about them having the span). Sorry, for the wrong suggestion then.

Feel free to revert to `.from_expansion` or `.is_empty` or `!=` but keep some FIXME there, then r=me